### PR TITLE
feat: adding cloud event structured logging sample

### PR DESCRIPTION
**Pull request by pattishin**
Structured logging http for GCFv2

**To run sample:**
1)  You will need to create a pubsub topic to test this sample. 
`gcloud pubsub topics create YOUR_TOPIC_NAME `

1) Execute below commands in either local terminal or cloud shell.
```
npm i
gcloud beta functions deploy nodejs-pubsub-function --gen2 --runtime nodejs16 --trigger-topic YOUR_TOPIC_NAME --entry-point structuredLoggingEvent --source . --project YOUR_PROJ
```

2) After successful deployment, trigger the function w/ empty message.
`gcloud pubsub topics publish my-logging-topic --message=""`

3) Open up logs and verify `Hello, world!` has logged and that you are able to see something similar to below.

Screen Shot 2022-04-11 at 9 53 23 AM

4) Run unit test via `npm run test`

### CODE REVIEW

**Comment by pattishin** [comment]
Checking out the linting/region tag errors!

File: functions/v2/cloud-event-logging/.gcloudignore
Comment (averikitsch) [inline functions/v2/cloud-event-logging/.gcloudignore:1]: same as other PR

File: functions/v2/cloud-event-logging/index.js
Comment (averikitsch) [inline functions/v2/cloud-event-logging/index.js:18]: Same comments apply to this PR as well.

File: functions/v2/cloud-event-logging/index.js
Comment (averikitsch) [inline functions/v2/cloud-event-logging/index.js:33]: The goal of this sample is to show "syntactic sugar" and provide a separate sample for code that doesn't have a request object. cloudEvent is not parsed by the library like httpRequest is for trace propagate. Though this brings up an interesting idea of adding all the cloud event info to the log may be useful.
```
const entry = log.entry({"arbitrary": "field"}, "Hello, world!");
log.error(entry)
```

File: functions/v2/cloud-event-logging/package.json
Comment (averikitsch) [inline functions/v2/cloud-event-logging/package.json:9]: Google LLC and Apache-2.0

File: functions/v2/cloud-event-logging/test/index.test.js
Comment (averikitsch) [inline functions/v2/cloud-event-logging/test/index.test.js:23]: Same advice as other PR

**Review by averikitsch — CHANGES_REQUESTED** [review]

**Comment by averikitsch** [comment]
I don't see any v2 tests - so we will have to add something like https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/.github/workflows/functions-http.yaml

File: functions/v2/cloud-event-logging/index.js
Comment (iennae) [inline functions/v2/cloud-event-logging/index.js:34]: @pattishin does this look like a linting issue you should fix?

Comment (pattishin) [reply functions/v2/cloud-event-logging/index.js:34]: sure does :)

File: functions/v2/cloud-event-logging/index.js
Comment (ace-n) [inline functions/v2/cloud-event-logging/index.js:28]: Should this be extracted out into a variable? e.g:

```
// TODO<developer>: specify your project ID here
const projectId = "MY_PROJECT";
...
await logging.setProjectId(projectId);
```

(A la [this example Node.js snippet](https://googlecloudplatform.github.io/samples-style-guide/#arrange-act-assert).)

File: functions/v2/cloud-event-logging/index.js
Comment (ace-n) [inline functions/v2/cloud-event-logging/index.js:26]: Can we hard-code this instead of importing `package.json`?

Assuming this is the name of a group of log entries, we might do something like this:
```
// Logger name to group log entries under
const loggerName = "my-logger";
...
const log = logging.logSync(loggerName)
```

File: functions/v2/cloud-event-logging/test/index.test.js
Comment (ace-n) [inline functions/v2/cloud-event-logging/test/index.test.js:21]: Ditto - can we do without `package.json` imports here?

File: functions/v2/cloud-event-logging/test/index.test.js
Comment (ace-n) [inline functions/v2/cloud-event-logging/test/index.test.js:35]: This comment is probably extraneous; our tests (and CI) often assume these are specified as global environment variables.

File: functions/v2/cloud-event-logging/test/index.test.js
Comment (ace-n) [inline functions/v2/cloud-event-logging/test/index.test.js:53]: Do we want to assert the following, or is this overkill? (Those "step by step" details can be helpful when debugging on CI.)
* `JSON.parse` of `process.stdout.write.getCall(0).args[0]` succeeds
* `process.stdout.write` was called (at least) once
* `process.stdout.write.getCall(0).args` is non-empty

**Review by ace-n — CHANGES_REQUESTED** [review]

File: functions/v2/cloud-event-logging/test/index.test.js
Comment (pattishin) [reply functions/v2/cloud-event-logging/test/index.test.js:62]: Sure, I can add a few more assertions

File: functions/v2/cloud-event-logging/index.js
Comment (averikitsch) [reply functions/v2/cloud-event-logging/index.js:27]: It's very common for the logger name to be the package name. I believe this fits recommended practices better than a place holder variable

File: functions/v2/cloud-event-logging/index.js
Comment (averikitsch) [inline functions/v2/cloud-event-logging/index.js:23]: Remove. The environment will provide this info

File: functions/v2/cloud-event-logging/index.js
Comment (averikitsch) [inline functions/v2/cloud-event-logging/index.js:29]: remove projectId variable

File: functions/v2/cloud-event-logging/index.js
Comment (averikitsch) [inline functions/v2/cloud-event-logging/index.js:33]: Let's remove severity here to show that severity can be set like: `log.error(entry)`

File: functions/v2/cloud-event-logging/index.js
Comment (ace-n) [reply functions/v2/cloud-event-logging/index.js:27]: I'm assuming this is a Cloud Logging practice more than a Node-specific one.

In that case, SGTM.

File: functions/v2/cloud-event-logging/test/index.test.js
Comment (ace-n) [reply functions/v2/cloud-event-logging/test/index.test.js:22]: Resolved per Averi's comment (on the other thread).

**Review by ace-n — APPROVED** [review]

**Comment by ace-n** [comment]
LGTM once @averikitsch approves. 👍

File: functions/v2/cloud-event-logging/index.js
Comment (pattishin) [reply functions/v2/cloud-event-logging/index.js]: Sounds good 👍🏼

**Review by averikitsch — APPROVED** [review]

**Comment by averikitsch** [comment]
@sofisl PTAL

**Pull request merged**
